rfcs/20230807-unify-primitive-and-graph-verbose/README.md

```diff
@@ -0,0 +1,147 @@
+interface `ONEDNN_GRAPH_VERBOSE` shown below for users to profile an application
+or troubleshoot API usage errors.
+
+| **Value**       | **Description**                                              |
+| --------------- | ------------------------------------------------------------ |
+| 0               | No verbose output                                            |
+| 1(profile_exec) | Information of compiled partition  execution                 |
+| 2(profile)      | Information of partition compilation and  compiled partition execution |
+| profile_compile | Information of partition compilation                         |
+| check           | Information of some verifying info                           |
+| error           | Information of some runtime error                            |
+| all             | All information above                                        |
+
+Take `profile` as an example, users can collect the profiling information of
+partition compilation and compiled partition execution by specifying
+`ONEDNN_GRAPH_VERBOSE=profile`. Verbose log containing marker, operation,
+engine_kind, partition_id, partition_kind, op_names, data_formats,
+logical_tensors, fpmath_mode, backend_name, time will be printed in the similar
+format as oneDNN primitive verbose.
+
+```bash
+CMD: ONEDNN_GRAPH_VERBOSE=profile ./tests/benchdnn/benchdnn --graph --case=pattern/f32/conv_bias_add_fusion.json
+LOG:
+onednn_graph_verbose,info,oneDNN v3.2.0 (commit ce0e4a11be3fa932bdc24f67dceff4df134d0207)
+onednn_graph_verbose,info,cpu,runtime:OpenMP,nthr:128
+onednn_graph_verbose,info,cpu,isa:Intel AVX-512 with Intel DL Boost
+onednn_graph_verbose,info,gpu,runtime:none
+onednn_graph_verbose,info,backend,0:dnnl_backend
+onednn_graph_verbose,compile:cache_miss,cpu,100006,convolution_post_ops,CONV_0;BINARY_1,data:NCX; filter:OIX;;,in0_f32:0:strided:undef:50x64x56x56:200704s3136s56s1 in1_f32:1:strided:constant:64x64x1x1:64s1s1s1 in2_f32:2:strided:constant:64:1 in3_f32:3:strided:undef:1x1x1x1:1s1s1s1 out0_f32:2209:strided:undef:50x64x56x56:200704s3136s56s1,fpm:strict,dnnl_backend,66.7661
+onednn_graph_verbose,exec,cpu,100006,convolution_post_ops,CONV_0;BINARY_1,data:NCX; filter:OIX;;,in0_f32:0:strided:undef:50x64x56x56:200704s3136s56s1 in1_f32:1:strided:constant:64x64x1x1:64s1s1s1 in2_f32:2:strided:constant:64:1 in3_f32:3:strided:undef:1x1x1x1:1s1s1s1 out0_f32:2209:strided:undef:50x64x56x56:200704s3136s56s1,fpm:strict,dnnl_backend,18.824
+```
+
+## Motivation
+
+oneDNN Graph API feature has been enabled and built by default in oneDNN's build
+system and it is general available since oneDNN v3.1 release. From the user
+perspective, a single unified verbose interface can be easier and
+straightforward to control the verbose mode for both primitive and graph APIs.
+From the implementation perspective, unifying them will save lots of duplicated
+codes in the graph API implementation and it will be better to maintain and
+align with the changes from primitive verbose.
```

Comment:
One thing to keep in mind, is that users might want to keep those two verbose output separate.
In particular, if a user uses only the graph API, do we want them to get the verbose output from both graph and primitive APIs?

Regarding simplifying internal implementation, that makes sense. I guess this could be achieved independently of the exposed knob though (e.g. we could use a single set of macros/messages but with separated verbose flags).

Reply:
yes. this proposal adds new env flags to allow users get primitive/graph verbose only.

Reply:
to avoid blowing up the number of verbose flags, `filter` has been proposed in this RFC. Please take a review, thanks:)